src/MSBuild/XMake.cs

```diff
@@ -1901,25 +1911,27 @@ private static void GatherResponseFileSwitch(string unquotedCommandLineArg, Comm
                             }
                         }
 
-                        GatherCommandLineSwitches(argsFromResponseFile, commandLineSwitches);
+                        commandLine += $"\n{responseFile}: {string.Join(" ", argsFromResponseFile)}";
```

Comment:
Ah, now I see why the command line string has to be passed by ref. I'm still not sure if this is the best approach. Ideally the command line would be kept in a more structured format (e.g. a dictionary mapping "response file" to "args from the response file") and maybe in a static field so it doesn't have to be passed around everywhere.

Aside from nicer and more allocation friendly code, It would make it possible to have a richer format of the output. I'm having second thoughts about the bare `<response file>: <command line>`, whether it is clear/expressive enough.